gordo_components/model/anomaly/diff.py

```diff
@@ -0,0 +1,152 @@
+            Used for transforming model output and the original ``y`` to calculate
+            the difference/error in model output vs expected.
+        """
+        self.base_estimator = base_estimator
+        self.scaler = scaler
+
+    def __getattr__(self, item):
+        """
+        Treat this model as transparent into base_estimator unless
+        referring to something owned by this object
+        """
+        if item in self.__dict__:
+            return getattr(self, item)
+        else:
+            return getattr(self.base_estimator, item)
+
+    def get_metadata(self):
+        if isinstance(self.base_estimator, GordoBase):
+            return self.base_estimator.get_metadata()
+        else:
+            return {}  # TODO: Fill this in.
+
+    def score(
+        self,
+        X: Union[np.ndarray, pd.DataFrame],
+        y: Union[np.ndarray, pd.DataFrame],
+        sample_weight: Optional[np.ndarray] = None,
+    ) -> float:
+        return self.base_estimator.score(X, y)
+
+    def save_to_dir(self, directory: str):
+        serializer.dump(self.base_estimator, os.path.join(directory, "base_estimator"))
+        serializer.dump(self.scaler, os.path.join(directory, "scaler"))
+        with open(os.path.join(directory, "params.json"), "w") as f:
+            params = {
+                k: v
+                for k, v in self.get_params().items()
+                if k not in ("base_estimator", "scaler")
+            }
+            json.dump(params, f)
+
+    @classmethod
+    def load_from_dir(cls, directory: str):
+        with open(os.path.join(directory, "params.json"), "r") as f:
+            params = json.load(f)
+        params["base_estimator"] = serializer.load(
+            os.path.join(directory, "base_estimator")
+        )
+        params["scaler"] = serializer.load(os.path.join(directory, "scaler"))
+        return cls(**params)
+
+    def get_params(self, deep=True):
+        return {"base_estimator": self.base_estimator, "scaler": self.scaler}
+
+    def fit(self, X: np.ndarray, y: np.ndarray):
+        self.base_estimator.fit(X, y)
+        self.scaler.fit(y)  # Scaler is used for calculating errors in .anomaly()
+        return self
+
+    def anomaly(
+        self, X: pd.DataFrame, y: pd.DataFrame, frequency: Optional[timedelta] = None
+    ) -> pd.DataFrame:
+        """
+        Create an anomaly dataframe from the base provided dataframe.
+
+        Parameters
+        ----------
+        X: pd.DataFrame
+            Dataframe representing the data to go into the model.
+        y: pd.DataFrame
+            Dataframe representing the target output of the model.
+
+        Returns
+        -------
+        pd.DataFrame
+            A superset of the original base dataframe with added anomaly specific
+            features
+        """
+
+        # Get the model output, falling back to transform if 'predict' doesn't exist
+        model_output = self.predict(X) if hasattr(self, "predict") else self.transform
```

Comment:
This is definitely nicer than the try-catch, even though the try-catch might be regarded as more [pythonic](https://docs.quantifiedcode.com/python-anti-patterns/readability/asking_for_permission_instead_of_forgiveness_when_working_with_files.html)